src/wayland/handlers/screencopy/render.rs

```diff
@@ -113,7 +114,8 @@ where
             }
             Ok(())
         })
-        .unwrap()
+        .map_err(|err| <R as Renderer>::Error::from_gles_error(GlesError::BufferAccessError(err)))
+        .and_then(|x| x)
         {
             frame.fail(FailureReason::Unknown);
```

Comment:
Hmm, it is debatable if this shouldn't be a client constraints error instead.

Reply:
Actually, `frame.fail` may not really be needed for the `BufAccessError`, since Smithay sends a `wl_buffer` protocol error when that fails.

Reply:
Well then it doesn't matter, because the client dies anyway. Better to send this rather than not imo.